packages/shopify-api/lib/index.ts

```diff
@@ -48,22 +44,20 @@ export interface Shopify<
 }
 
 export function shopifyApi<
-  Params extends ConfigParams<any, Future>,
-  Resources extends DefaultedResources<Params['restResources']>,
+  Params extends ConfigParams<Resources, Future>,
+  Resources extends ShopifyRestResources,
   Future extends FutureFlagOptions,
 >({
   future,
+  restResources,
   ...config
-}: {future?: Future} & Params): Shopify<
+}: {future?: Future; restResources?: Resources} & Params): Shopify<
```

Comment:
Why is this change a future flag? I assume it's not a breaking change, is it just that we want to eventually remove some code?

Reply:
This is actually not a future flag, but there were some types here that were a bit messy, and I'd already had to spend some time fixing them before rolling back. I figured we could keep the fixes around - this is mostly for the benefit of improving `loadRestResources`.